Development docs update (#480)
giantswarm/happa: README.md

# Happa

The Giant Swarm web user interface. It lets users:

- View and manage clusters
- Manage their account
- Add / remove organizations
- Add / remove members from organizations
- Learn how to get started with their kubernetes clusters by following a guide

Happa is intended to be deployed to Giant Swarm control planes and acts as a client to the Giant Swarm API.

![Screenshot of Happa](https://user-images.githubusercontent.com/455309/51164968-a5320780-18d9-11e9-91c5-10ad144d7ada.png)

Happa is a single page JavaScript application using React and runs in modern browsers.

## Getting started with development / demoing

Happa has to be configured for an API endpoint. There are two alternative methods
when running happa locally:

- (A) Bring up the API suite locally. See [giantswarm/api/](https://github.com/giantswarm/api/tree/master/testing) for details.
- (B) Modify the base config in [index.html](https://github.com/giantswarm/happa/blob/master/src/index.html) to point to a test installation.

### For (A) – Bring up the API suite locally

This requires Docker and `docker-compose`.

> Start the `api` and dependencies first by going to the `api` repo and running

Reviewer: ```suggestion
Start the `api` and dependencies first by going to the `api` [repo](https://github.com/giantswarm/api) and running
```

the dockercompose file there:

```nohighlight
cd $GOPATH/src/github.com/giantswarm/api/testing
make up
```

As part of `make up`, `./fixtures.sh` will run and create the initial user and
organization you can log in with.

### For (B)

> Find the `apiEndpoint` setting and set it to the API endpoit of the test

Reviewer: ```suggestion
Find the `apiEndpoint` setting in `src/index.html` and set it to the API endpoint of the test
```

installation you want to use.

If you want to test password resetting and other user account transactions,

> also adapt `passageEndpoint`.

Reviewer: ```suggestion
also adapt `passageEndpoint` value.
```
